contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+        );
+    }
+
+    /// @notice Calculates the current segment of the game.
+    /// @return current game segment
+    function getCurrentSegment() public view returns (uint256) {
+        return block.timestamp.sub(firstSegmentStart).div(segmentLength);
+    }
+
+    /// @notice Checks if the game is completed or not.
+    /// @return "true" if completeted; otherwise, "false".
+    function isGameCompleted() public view returns (bool) {
+        // Game is completed when the current segment is greater than "lastSegment" of the game.
+        return getCurrentSegment() > lastSegment;
+    }
+
+    function _transferDaiToContract(uint256 _minAmount) internal {
+        require(
+            daiToken.allowance(msg.sender, address(this)) >= segmentPayment,
+            "You need to have allowance to do transfer DAI on the smart contract"
+        );
+
+        uint256 currentSegment = getCurrentSegment();
+        players[msg.sender].mostRecentSegmentPaid = currentSegment;
+        players[msg.sender].amountPaid = players[msg.sender].amountPaid.add(
+            segmentPayment
+        );
+        // check if this is deposit for the last segment. If yes, the player is a winner.
+        // since both join game and deposit method call this method so having it here
+        if (currentSegment == lastSegment.sub(1)) {
+            winners.push(msg.sender);
+            // array indexes start from 0
+            players[msg.sender].winnerIndex = winners.length.sub(uint256(1));
+            winnerCount = winnerCount.add(uint256(1));
+            players[msg.sender].isWinner = true;
+        }
+
+        totalGamePrincipal = totalGamePrincipal.add(segmentPayment);
+        require(
+            daiToken.transferFrom(msg.sender, address(this), segmentPayment),
+            "Transfer failed"
+        );
+
+        // Allows the lending pool to convert DAI deposited on this contract to aDAI on lending pool
+        uint256 contractBalance = daiToken.balanceOf(address(this));
+        require(
+            daiToken.approve(address(pool), contractBalance),
+            "Fail to approve allowance to pool"
+        );
+
+        if (poolType == 0) {
+            uint256[numAaveTokens] memory amounts;
+            for (uint256 i = 0; i < numAaveTokens; i++) {
+                if (i == inboundTokenIndexUint) {
+                    amounts[i] = segmentPayment;
+                } else {
+                    amounts[i] = 0;
+                }
+            }
+            pool.add_liquidity(amounts, _minAmount, true);
+        } else if (poolType == 1) {
+            uint256[numAtricryptoTokens] memory amounts;
+            for (uint256 i = 0; i < numAtricryptoTokens; i++) {
+                if (i == inboundTokenIndexUint) {
+                    amounts[i] = segmentPayment;
+                } else {
+                    amounts[i] = 0;
+                }
+            }
+            pool.add_liquidity(amounts, _minAmount);
+        }
```

Comment:
this can be simplified. you can use the if condition to the `numAaveTokens` or `numAtricryptoTokens` to a variable and run the loop using that. Everything else seems the same.

Then use the `if` condition to call `add_liquidity` accordingly.

Reply:
numAaveTokens has to be a constant type actually other wise the signature becomes diff. and the external call will fail if I use a "if" condition the assignment will be to a non-constant ver, this again is due to the structure of how the curve contracts are written